apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react';
+import styles from './InPageNavigation.module.scss';
+
+const useInPageNavigation = () => {
+    useEffect(() => {
+        const sections = Array.from(document.querySelectorAll('section[id]'));
+
+        const scrollHandler = (entries: IntersectionObserverEntry[]) => {
+            entries.forEach((entry) => {
+                const section = entry.target as HTMLElement;
+                const sectionId = section.id;
+                const sectionLink = document.querySelector(`a[href="#${sectionId}"]`);
+                if (entry.intersectionRatio > 0) {
```

Comment:
suggestion (non-blocking): we could use a prop here to customize the value intersectionRatio is compared against in future implementations of inPageNavigation.